### PR TITLE
dnn: use OpenVINO 2021.4 defines

### PR DESCRIPTION
**Pull request by alalek**
https://github.com/openvinotoolkit/openvino/releases/tag/2021.4

previous update #19770

```
force_builders_only=Custom
build_image:Custom=ubuntu-openvino-2021.4.0:20.04
build_image:Custom Win=openvino-2021.4.0
build_image:Custom Mac=openvino-2021.4.0

test_modules:Custom=dnn,python2,python3,java
test_modules:Custom Win=dnn,python2,python3,java
test_modules:Custom Mac=dnn,python2,python3,java

buildworker:Custom=linux-1
test_opencl:Custom=OFF
test_bigdata:Custom=1
test_filter:Custom=*
YOLO*:*VINO*:*Infer*:*Layer*:*layer*

build_contrib:Custom Win=OFF
build_examples:Custom Win=OFF

allow_multiple_commits=1
```

### CODE REVIEW

File: modules/dnn/src/ie_ngraph.cpp [hunk 1]
```diff
@@ -1002,35 +1006,53 @@ void InfEngineNgraphNet::forward(const std::vector<Ptr<BackendWrapper> >& outBlo
...
+        InferenceEngine::IInferRequest& infRequest = *infRequestPtr.get();
+        infRequest.SetUserData(reqWrapper.get(), 0);
+#endif
 
-        infRequestPtr->SetCompletionCallback(
-            [](InferenceEngine::IInferRequest::Ptr request, InferenceEngine::StatusCode status)
+#if INF_ENGINE_VER_MAJOR_GE(INF_ENGINE_RELEASE_2021_4)
+        infRequest.SetCompletionCallback<std::function<void(InferenceEngine::InferRequest, InferenceEngine::StatusCode)>>(
+            [reqWrapper](InferenceEngine::InferRequest /*request*/, InferenceEngine::StatusCode status)
+#else
```
Comment (alalek) [inline modules/dnn/src/ie_ngraph.cpp:1021]: @dkurt @ilya-lavrenov Could you please take a look on this replacement of deprecated `IInferRequest`?

Comment (dkurt) [reply modules/dnn/src/ie_ngraph.cpp:1021]: Looks good as for me

File: modules/dnn/src/ie_ngraph.cpp [hunk 2]
```diff
@@ -1002,35 +1006,53 @@ void InfEngineNgraphNet::forward(const std::vector<Ptr<BackendWrapper> >& outBlo
...
+        CV_Assert(infRequestPtr);
+        InferenceEngine::IInferRequest& infRequest = *infRequestPtr.get();
+        infRequest.SetUserData(reqWrapper.get(), 0);
+#endif
 
-        infRequestPtr->SetCompletionCallback(
-            [](InferenceEngine::IInferRequest::Ptr request, InferenceEngine::StatusCode status)
+#if INF_ENGINE_VER_MAJOR_GE(INF_ENGINE_RELEASE_2021_4)
+        infRequest.SetCompletionCallback<std::function<void(InferenceEngine::InferRequest, InferenceEngine::StatusCode)>>(
+            [reqWrapper](InferenceEngine::InferRequest /*request*/, InferenceEngine::StatusCode status)
```
Comment (ilya-lavrenov) [inline modules/dnn/src/ie_ngraph.cpp:1020]: * callback here holds `reqWrapper`
* `infRequest` holds callback

We can have multiple problems with this. We don't need to hold request inside callback

Comment (alalek) [reply modules/dnn/src/ie_ngraph.cpp]: Thank you! Replaced to raw pointer in callback.

**Comment by alalek** [comment]
:+1:

**Pull request merged**